### PR TITLE
RHBA-499: Add documentation for project migration tool.

### PR DESCRIPTION
**Pull request by mbarkley**

### CODE REVIEW

File: kie-wb-common-cli/kie-wb-common-project-migration/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+* `$NIOGIT` is the path of your `.niogit` directory being migrated.
+* `$TOOL_DIR` is the path to the directory installed in the previous section (i.e. `$INSTALL_DIR/kie-wb-common-project-migration-$VERSION/`).
+
+Simple Usage
+------------
+
+The simplest way to invoke the tool is
+
+    $TOOL/bin/migration-tool.sh -t $NIOGIT
```
Comment (tomasdavidorg) [inline kie-wb-common-cli/kie-wb-common-project-migration/README.md]: $TOOL should be replaced with $TOOL_DIR

File: kie-wb-common-cli/kie-wb-common-project-migration/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Batch Mode
+----------
+
+The simple invocation will prompt the user to confirm before migration is attempted. This is done to warn users about making backups, in case an error occurs while migrating project data.
+
+If you wish to run the tool without this prompt, you can add the `-b` flag like so:
+
+
+    $TOOL/bin/migration-tool.sh -b -t $NIOGIT
```
Comment (tomasdavidorg) [inline kie-wb-common-cli/kie-wb-common-project-migration/README.md]: Same here.

File: kie-wb-common-cli/kie-wb-common-project-migration/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+For the instructions below, these variables are assumed:
+
+* `$NIOGIT` is the path of your `.niogit` directory being migrated.
+* `$TOOL_DIR` is the path to the directory installed in the previous section (i.e. `$INSTALL_DIR/kie-wb-common-project-migration-$VERSION/`).
+
+Simple Usage
+------------
+
+The simplest way to invoke the tool is
```
Comment (tomasdavidorg) [inline kie-wb-common-cli/kie-wb-common-project-migration/README.md]: What are the steps after running this migration script? Should be the new .niogit copied to the new workbench? Could you please extend this part?

File: kie-wb-common-cli/kie-wb-common-project-migration/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+To install the tool, simply unzip it into the desired location, like so:
+
+    unzip -d $INSTALL_DIR kie-wb-common-project-migration-$VERSION-linux.zip
+
+This should create a directory (`kie-wb-common-project-migration-$VERSION`) in the `$INSTALL_DIR`.
+
+Using the Tool
+--------------
+
+The tool is meant to operate on the `.niogit` directory of a workbench distribution. *IMPORANT*: there should not be a running workbench using this directory when using the tool.
```
Comment (mareknovotny) [inline kie-wb-common-cli/kie-wb-common-project-migration/README.md]: typo in IMPORANT

File: kie-wb-common-cli/kie-wb-common-project-migration/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+Maven Build
+-----------
+
+To build the project, simply run
+
+    mvn clean install
+
+This will generate a zip file in the target directory:
+
+    kie-wb-common-project-migration-$VERSION-linux.zip
```
Comment (triceo) [inline kie-wb-common-cli/kie-wb-common-project-migration/README.md]: How about our Windows customers?

File: kie-wb-common-cli/kie-wb-common-project-migration/README.md (hunk 3 above)
Comment (mbarkley) [reply kie-wb-common-cli/kie-wb-common-project-migration/README.md]: Added a description.

**Comment by mareknovotny** [comment]
@mbarkley what do you think about missing windows launcher?

**Comment by mbarkley** [comment]
@triceo @mareknovotny We certainly need it. I just don't know how to script on Windows or have an environment ready to test this.

There's an [issue](https://issues.jboss.org/browse/RHBA-514) open for this now. If there are no other volunteers I can figure it out, but it would be faster if we had a volunteer who is more familiar with Windows.

It isn't a lot of work. The script just needs to launch Java with the appropriate classpath (all the jars in the lib directory). Once this is added we can probably bundle it in the same zip file and get rid of the "linux" qualifier in the zip file name.

**Comment by mareknovotny** [comment]
@mbarkley that's fair and thanks for pointing to the existing jira issue

**Comment by mareknovotny** [comment]
our CI is too sensitive, even update of README can fail tests :)

**Pull request merged**